### PR TITLE
[6707] Validate macros in Content Type form before saving

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6707
https://github.com/craftercms/craftercms/issues/6705

### CODE REVIEW

**Comment by jvega190** [comment]
@rart I added the validation in the content-type editor, in the input for destination path pattern. I wonder if this should go in another level, and also how to display the error to the user. I'll ping you to discuss.

Update: I'll also work on this ticket https://github.com/craftercms/craftercms/issues/6705 together with this one.

**Review by rart — COMMENTED** [review]
Pls call to demo, review and discuss

File: static-assets/components/cstudio-admin/mods/content-types.js [hunk 1]
```diff
@@ -2300,6 +2300,28 @@
...
+                invalidMacrosInPath.push(macro);
+                el.value = el.value.replace(regex, '');
+              }
+            });
+            if (invalidMacrosInPath.length > 0) {
+              CStudioAuthoring.Utils.showNotification(
+                formatMessage(contentTypesMessages.invalidMacros, { macros: invalidMacrosInPath.join(', ') }),
+                'top',
+                'left',
+                'success',
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/content-types.js]: This should be an info.

File: ui/app/src/env/i18n-legacy.ts [hunk 2]
```diff
@@ -310,6 +310,9 @@ export const contentTypesMessages = defineMessages({
   },
   audioManager: {
     defaultMessage: 'Audio Manager'
+  },
+  invalidMacros: {
+    defaultMessage: 'Path contains invalid macros: {macros}'
```
Comment (rart) [inline ui/app/src/env/i18n-legacy.ts]:
```suggestion
    defaultMessage: 'The following invalid macros were removed: {macros}'
```

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**